runtime/oti/j9.h

```diff
@@ -339,6 +339,10 @@ static const struct { \
 #define J9_IS_FIELD_FLATTENED(fieldClazz, romFieldShape) \
 		(J9_IS_J9CLASS_FLATTENED(fieldClazz) && \
 		(J9_ARE_NO_BITS_SET((romFieldShape)->modifiers, J9AccVolatile) || (J9CLASS_UNPADDED_INSTANCE_SIZE(fieldClazz) <= sizeof(U_64))))
+/* This will replace J9_IS_FIELD_FLATTENED when QTypes are removed */
+#define J9_IS_NULL_RESTRICTED_FIELD_FLATTENED(fieldClazz, romFieldShape) \
+		J9_ARE_ALL_BITS_SET((romFieldShape)->modifiers, J9FieldFlagIsNullRestricted) && \
```

Comment:
I see `J9_IS_J9CLASS_FLATTENED(fieldClazz)` is removed. After looking at the code closer, I think we should still keep  `J9_IS_J9CLASS_FLATTENED(fieldClazz)`, as user can use `-XX:ValueTypeFlatteningThreshold=<size>` to control whether `J9ClassIsFlattened` will be set on the J9Class based on the class size. The should change the field flattening behaviour.

Reply:
yes thats true. that addition will make it so J9_IS_NULL_RESTRICTED_FIELD_FLATTENED since its only true for primitive value classes.

Reply:
I have added J9_IS_NULL_RESTRICTED_FIELD_FLATTENED